iceoryx_utils/test/moduletests/test_posix_file_lock.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_utils/posix_wrapper/file_lock.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::posix;
+using namespace iox::cxx;
+
+constexpr char TEST_NAME[] = "TestProcess";
+constexpr char ANOTHER_TEST_NAME[] = "AnotherTestProcess";
+
+/// @req
+/// @brief This test suite verifies the behaviour of the class FileLock
```

Comment:
I'm happy to re-write the comment, trying not to be too verbose. However, we [need](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/aspice_swe3_4/example/iceoryx_component/test/moduletests/test_example_base_class.cpp#L29-L33) these doxygen comments in every unit test for linkage to the requirements.

Reply:
I do not think so. If we follow our guidelines then the linkage can be perfectly done via the filenames!

Reply:
I don't know if this will be possible for all cases. However, before I'll adapt anything, please raise a PR changing our [example](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/aspice_swe3_4/example/iceoryx_component/test/moduletests/test_example_base_class.cpp).